mono/sgen/sgen-memory-governor.c

```diff
@@ -164,11 +170,37 @@ void
 sgen_memgov_minor_collection_start (void)
 {
 	total_promoted_size_start = total_promoted_size;
+	SGEN_TV_GETTIME (last_minor_start);
+}
+
+static void
+sgen_add_log_entry (SgenLogEntry *log_entry)
+{
+	mono_coop_mutex_lock (&log_entries_mutex);
+	sgen_pointer_queue_add (&log_entries, log_entry);
+	mono_coop_mutex_unlock (&log_entries_mutex);
 }
 
 void
-sgen_memgov_minor_collection_end (void)
+sgen_memgov_minor_collection_end (const char *reason, gboolean is_overflow)
 {
+	if (mono_trace_is_traced (G_LOG_LEVEL_INFO, MONO_TRACE_GC)) {
+		SgenLogEntry *log_entry = (SgenLogEntry*)sgen_alloc_internal_dynamic (sizeof (SgenLogEntry), INTERNAL_MEM_TEMPORARY, TRUE);
```

Comment:
Please add an internal memory type for log entries and use that.  Also, it's fixed size, so no need to use dynamic.